io/src/main/scala/sbt/io/IO.scala

```diff
@@ -373,19 +378,17 @@ object IO {
   }
 
   /** Returns the children of directory `dir` that match `filter` in a non-null array.*/
-  def listFiles(filter: java.io.FileFilter)(dir: File): Array[File] = wrapNull(dir.listFiles(filter))
+  def listFiles(filter: java.io.FileFilter)(dir: File): Vector[File] = wrapNull(dir.listFiles(filter))
```

Comment:
Array => Vector is definitely not an improvement here.  We actually want to move to something even lazier, if we can.  See comments below.